flink-rpc/flink-rpc-akka/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcService.java

```diff
@@ -137,6 +138,13 @@ public AkkaRpcService(
 
         captureAskCallstacks = configuration.captureAskCallStack();
 
+        // Akka always sets the threads context class loader to the class loader with which it was
+        // loaded (i.e., the plugin class loader)
+        // we must ensure that the context class loader is set to the Flink class loader when we
+        // call into Flink
+        // otherwise we could leak the plugin class loader or poison the context class leader of
+        // external threads (because they inherit the current threads context class loader)
+        internalExecutor = new ContextClassLoaderCleaningExecutor(actorSystem.dispatcher());
```

Comment:
The akka submodule usually lives as long as the cluster, right?
So most of the changes in this commit are more for the MiniCluster, such that JVM can outlive the submodule for a specific cluster.
I'd quickly like to discuss if the benefit outweighs the cost. For the cost, it pretty much means that for all RPC calls, we switch the context cl 3 times (before call: akka switches, we switch, after call: we switch). Did we ever measure the performance of an RPC call and have an idea if this is back-and-forth is noticeable? The benefit would be proper isolation in tests. I tend to think that it's worth it but I have no idea on how many RPC calls there are and how heavy they are beyond this switcheroo.

Reply:
its not at all about the lifecycle; please see the PR description, it does into detail why this is important.

Reply:
I have not measured the performance impact, but I would doubt that it is significant.

Reply:
That's good enough for me. I guess RPC calls are just comparably rare compared to everything on the hot path.